src/pool-service/IrcClientRedisState.ts

```diff
@@ -21,20 +36,50 @@ interface IrcClientStateDehydrated {
     capabilities: ReturnType<IrcCapabilities["serialise"]>;
     supportedState: IrcSupported;
     hostMask: string;
-    chans: [string, ChanData][];
+    chans: [string, ChanDataDehydrated][];
     prefixForMode: {
         [mode: string]: string;
     };
     maxLineLength: number;
     lastSendTime: number;
 }
 
+
 export class IrcClientRedisState implements IrcClientState {
-    private putStatePromise: Promise<void> = Promise.resolve();
+    private putStatePromise: Promise<unknown> = Promise.resolve();
 
-    static async create(redis: Redis, clientId: string) {
-        const data = await redis.hget(REDIS_IRC_CLIENT_STATE_KEY, clientId);
+    static async create(redis: Redis, clientId: string, freshState: boolean) {
+        log.debug(`Requesting ${freshState ? "fresh" : "existing"} state for ${clientId}`);
+        const data = freshState ? null : await redis.hget(REDIS_IRC_CLIENT_STATE_KEY, clientId);
         const deseralisedData = data ? JSON.parse(data) as IrcClientStateDehydrated : {} as Record<string, never>;
+        const chans = new Map<string, ChanData>();
+
+        // In a previous iteration we failed to seralise this properly.
+        const isPreviousBuggyState = !Array.isArray(deseralisedData.chans?.[0]?.[1]?.users);
+        deseralisedData?.chans?.forEach(([channelName, chanData]) => {
+            chans.set(channelName, {
+                ...chanData,
+                users: new Map(!isPreviousBuggyState ? chanData.users : []),
+                modeParams: new Map(!isPreviousBuggyState ? chanData.modeParams : []),
+            })
+        });
+
+        // We also had a bug where the supported state is bloated enormously
+        if (deseralisedData.supportedState) {
+            deseralisedData.supportedState = {
+                ...deseralisedData.supportedState,
+                channel: {
+                    ...deseralisedData.supportedState.channel,
+                    modes: {
+                        a: [...new Set(deseralisedData.supportedState.channel.modes.a.split(''))].join(''),
+                        b: [...new Set(deseralisedData.supportedState.channel.modes.b.split(''))].join(''),
+                        c: [...new Set(deseralisedData.supportedState.channel.modes.c.split(''))].join(''),
+                        d: [...new Set(deseralisedData.supportedState.channel.modes.d.split(''))].join(''),
+                    }
+                },
+                extra: [...new Set(deseralisedData.supportedState.extra)],
+            }
+        }
```

Comment:
Is this more readable (as long as it's functionally equivalent)?
```suggestion
        if (deseralisedData.supportedState) {
            deseralisedData.supportedState.channel.modes = {
                a: [...new Set(deseralisedData.supportedState.channel.modes.a.split(''))].join(''),
                b: [...new Set(deseralisedData.supportedState.channel.modes.b.split(''))].join(''),
                c: [...new Set(deseralisedData.supportedState.channel.modes.c.split(''))].join(''),
                d: [...new Set(deseralisedData.supportedState.channel.modes.d.split(''))].join(''),
            }
            deseralisedData.supportedState.extra = [...new Set(deseralisedData.supportedState.extra)];
        }
```